packages/ComponentLibrary/src/components/TextInput/TextInput.types.ts

```diff
@@ -0,0 +1,10 @@
+import { TextFieldProps } from '@mui/material';
+
+export type TextInputProps = {
+    autoCompleteTexts?: string[];
+    fetchSuggestions?: (query: string) => Promise<void>;
+    leftIcon?: React.ReactNode;
+    rightIcon?: React.ReactNode;
+    onLeftIconClick?: () => void;
+    onRightIconClick?: () => void;
+} & TextFieldProps;
```

Comment:
Is this equivalent to extending the class in typescript? If not, we would have to see how to extend it